pkg/server/ptypes/pipeline.go

```diff
@@ -51,6 +51,30 @@ func TestPipeline(t testing.T, src *pb.Pipeline) *pb.Pipeline {
 	return src
 }
 
+// TestPipelineStep creates a valid pipeline step for testing, with optional
+// workspace reference.
+func TestPipelineAppendSteps(t testing.T, src *pb.Pipeline, srcSteps []*pb.Pipeline_Step) *pb.Pipeline {
+	t.Helper()
+
+	pipe := TestPipeline(t, src)
+
+	for _, step := range srcSteps {
+		require.NoError(t, mergo.Merge(step, &pb.Pipeline_Step{
+			Name:      "next",
```

Comment:
The `Merge` only adds/changes values if the destination's value is empty, so only if `Name` doesn't already have a value

Reply:
That makes sense! Should this maybe be like `fmt.Println("next-%d", i)`? Otherwise it's invalid to have a pipeline where more than 1 step has the same name

Reply:
Chatted about this via slack. It's probably ok to leave this as is for now! I'll resolve.